module_ctx.go

```diff
@@ -737,6 +741,25 @@ func (mctx *mutatorContext) Rename(name string) {
 	mctx.rename = append(mctx.rename, rename{mctx.module.group, name})
 }
 
+// Create a new module by calling the factory method for the specified moduleType.  The module
+// will be empty, as if it were created by a blueprint file definition that contained no properties.
+// The caller should use AppendProperties
+func (mctx *mutatorContext) CreateModule(moduleType string, props ...interface{}) {
+	module := mctx.context.newModule(moduleType)
+	if module == nil {
+		panic(fmt.Errorf("Unknown module type %q", moduleType))
```

Comment:
Should we take into account ignoreUnknownModuleTypes? I'm leaning towards no, but it could be useful.